### PR TITLE
feat: simulate middle click when opening in a new background tab

### LINKED ISSUE #496: [Feature Request] Mimic middle-click on "Open in a new background tab"
Currently, "Open in a new background tab" opens a new tab in the same Firefox container as the link search page. I use the [Temporary Containers](https://addons.mozilla.org/en-US/firefox/addon/temporary-containers/) extension and have set my Isolation settings to open a new container on middle-click if the target belongs to a different domain or subdomain.

Is it possible to add an option to mimic the middle-click when the shortcut associated with "Open in a new background tab" is pressed?

I understand this may not be a popular feature, so I would be more than happy to start working on it myself and submit a PR. I would appreciate it if you could give me a few pointers on where to start.

I should probably try to modify one of the following blocks, right?
https://github.com/infokiller/web-search-navigator/blob/136bd59b5d8d5c64c94d8c80b6bc588ff56095f9/src/main.js#L510-L523
https://github.com/infokiller/web-search-navigator/blob/136bd59b5d8d5c64c94d8c80b6bc588ff56095f9/src/background.js#L1-L9

**Browser and OS**

- Browser: Firefox 111.0.1
- OS: macOS Ventura 13.2.1

### PR DESCRIPTION
**Pull request by MahdiNazemi**
Closes #496.

### CODE REVIEW

**Comment by infokiller** [comment]
Thanks for the contribution! I added one minor comment.
Did you test it manually, and does it resolve your issue?

**Comment by MahdiNazemi** [comment]
> Thanks for the contribution! I added one minor comment. Did you test it manually, and does it resolve your issue?

Thank _you_. Yes, I tested it in Firefox, the only browser Temporary Containers supports, and it worked as expected.

I have trouble finding your comment.

**Review by infokiller — CHANGES_REQUESTED** [review]

**Comment by infokiller** [comment]
> 



> > Thanks for the contribution! I added one minor comment. Did you test it manually, and does it resolve your issue?
> 
> Thank _you_. Yes, I tested it in Firefox, the only browser Temporary Containers supports, and it worked as expected.
> 
> I have trouble finding your comment.

Whoops, forgot to hit the "request changes" button. Now you should see it.
Thanks for testing!

File: src/main.js [hunk 1]
```diff
@@ -512,6 +512,30 @@ class WebSearchNavigator {
...
+        const middleClickMouseup = new MouseEvent('mouseup', {
+          bubbles: true,
+          cancelable: false,
+          view: window,
+          button: 1,
+          which: 2,
+          buttons: 0,
+          clientX: link.getBoundingClientRect().x,
+          clientY: link.getBoundingClientRect().y,
+        });
```
[the remark this answers is not in the document]
Comment (MahdiNazemi) [reply src/main.js:536]: Yes, the parameters are exactly the same, and your comment is valid.

Temporary Containers defines an event listener only on the `mouseup` event, and I had initially defined only that. Later, I also added `mousedown` to really mimic the middle click. Should I keep both and name the option middle click or only keep `mouseup` and name the option something like middle click release?

[the remark this answers is not in the document]
Comment (MahdiNazemi) [reply src/main.js]: Done!

**Comment by infokiller** [comment]
Looks great, merged it. I hope to release a new version to the stores (Chrome and Firefox) in the next few days.

**Pull request merged**